src/label.ts

```diff
@@ -9,16 +11,41 @@ export interface MatchExpression {
 
 export interface Selector {[l: string]: string | MatchExpression; }
 
-export function selectorToQueryString(selector: Selector): string {
-    const v: string[] = [];
-
-    _.forEach(selector, (value, label) => {
+export function selectorToString(selector: Selector, separator = ";"): string {
+    const parts = Object.entries(selector).map(([label, value]) => {
         if (typeof value === "string") {
-            v.push(label + "=" + value);
-        } else {
-            v.push(label + " "+value.operator+" (" + value.values.join(",") + ")");
+            return (label + "=" + value);
         }
+            return (label + " "+value.operator+" (" + value.values.join(",") + ")");
     });
 
-    return v.join(",");
+    return parts.join(separator);
+}
+
+export const selectorToQueryString = (selector: Selector) => selectorToString(selector, ",");
+
+/**
+ * Parse a Label Selector string to a Selector Object.
+ * Label Selectors are described in the [Kubernetes documentation](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors).
+ *
+ * Uses `;` as a separator for multiple expressions in the string.
+ */
+export function parseLabelSelector(input: string): Selector {
+    const selector: Selector = {};
+
+    for (const item of input.split(";")) {
+        const regex = new RegExp("([^\(\)]*) (in|notin) \\((.*)\\)", "i");
+        const regexResult = regex.exec(item);
+        if (regexResult !== null) {
+            selector[regexResult[1]] = {
+                operator: regexResult[2] as SetBasedOperator,
+                values: regexResult[3].split(","),
+            };
+        } else {
+            const [key, operator, ...values] = item.split(/(=|!=|==)/);
+            selector[key] = operator === "=" ? values.join('') : { operator: operator as EqualityBasedOperator, values };
```

Comment:
Should we also trim whitespaces from values? If someone uses `foo = bar` instead of `foo=bar` as selector?

Reply:
```typescript
    test("trims values if spaces are used", () => {
        const selector = parseLabelSelector("foo = bar;bar = baz");
        expect(selector).toEqual({foo: "bar", bar: "baz"});
    });
```
Here the trim is unproblematic, but should values also be trimmed in a match operation?
```typescript
    test("can handle whitespaces around values in match expression", () => {
        const selector = parseLabelSelector("scope in ( internal , external )");
        expect(selector).toEqual({
            scope: {
                operator: "in",
                values: ["internal", "external"]
            }
        });
    });
```

Reply:
As seen in the documentation this should be allowed
```
environment in (production, qa)
tier notin (frontend, backend)
```
so i’d trim these values as well.